pkg/controller/node/controller_utils.go

```diff
@@ -110,6 +111,10 @@ func deletePods(kubeClient clientset.Interface, recorder record.EventRecorder, n
 		if err == nil { // No error means at least one daemonset was found
 			continue
 		}
+		// if the pod is forgivable (wants to stay bound on the node), ignore it
+		if isPodForgivable(pod, nodeStatus.readyTransitionTimestamp) {
```

Comment:
deletePods will be called only if node is added to zonePodEvictor queue, it ensures the node is not ready.